firmware/thruna/FSM/Src/Io/Io_AcceleratorPedals.c

```diff
@@ -1,76 +1,127 @@
-#include <tgmath.h>
+//#include <tgmath.h>
+#include <math.h>
 #include "Io_AcceleratorPedals.h"
 #include "Io_Adc.h"
 #include "App_SharedMacros.h"
 
-// TODO: Change these lengths
+// TODO: Change these lengths (MM)
+// TODO: take SAPPS measurement
 // Constant used to compute gamma (cosine law)
-#define PAPPS_LEN_A (130.0f)
-#define PAPPS_LEN_B (117.0f)
+#define PAPPS_LEN_A (175.0f) // pedal length to PAPPS
+#define SAPPS_LEN_A (165.0f) // pedal length to SAPPS
+#define LEN_B (90.61f)       // horizontal distance to PAPPS/ SAPPS
+
 // Assume that the angle range is 50 degrees
-#define PAPPS_LENGTH_UNPRESSED_MM (199.0f)
-#define PAPPS_LENGTH_FULL_PRESSED_MM (152.0f)
+
+#define PAPPS_LENGTH_UNPRESSED_MM (210.0f)
+#define PAPPS_LENGTH_FULL_PRESSED_MM (160.0f) // remeasure
+
+#define SAPPS_LENGTH_UNPRESSED_MM (196.2f)
+#define SAPPS_LENGTH_FULL_PRESSED_MM (147.17f) // remeasure
+
 // Raw voltage seen at the pin
-#define PAPPS_FULL_PRESSED_POT_V (2.86f)
-#define PAPPS_UNPRESSED_POT_V (1.082f)
+
+#define PAPPS_FULL_PRESSED_POT_V (2.61f)
+#define PAPPS_UNPRESSED_POT_V (0.68f)
+
+#define SAPPS_FULL_PRESSED_POT_V (3.20f)
+#define SAPPS_UNPRESSED_POT_V (1.18f)
```

Comment:
Did you look into using the sensor datasheet to convert from voltage -> length?

Reply:
I did, I wasn't exactly sure if the independent linearity is the change in V/m so I ended up sticking to my experimental values.